examples/terraform/aws-dualstack/main.tf

```diff
@@ -0,0 +1,357 @@
+    (local.zoneB) = length(aws_subnet.public.*.id) > 0 ? aws_subnet.public[1].id : ""
+    (local.zoneC) = length(aws_subnet.public.*.id) > 0 ? aws_subnet.public[2].id : ""
+  }
+}
+
+################################# DATA SOURCES #################################
+
+data "aws_availability_zones" "available" {
+  state = "available"
+}
+
+data "aws_ami" "ami" {
+  most_recent = true
+  owners      = var.ami_filters[var.os].owners
+
+  filter {
+    name   = "name"
+    values = var.ami_filters[var.os].image_name
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+}
+
+data "aws_vpc" "selected" {
+  id = var.vpc_id
+}
+
+data "aws_internet_gateway" "default" {
+  filter {
+    name   = "attachment.vpc-id"
+    values = [data.aws_vpc.selected.id]
+  }
+}
+
+resource "random_integer" "cidr_block" {
+  min = 0
+  max = local.subnet_total - 1
+}
+
+############################### NETWORKING SETUP ###############################
+
+resource "aws_subnet" "public" {
+  count                   = 3
+  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  map_public_ip_on_launch = true
+  vpc_id                  = data.aws_vpc.selected.id
+
+  cidr_block = cidrsubnet(
+    data.aws_vpc.selected.cidr_block,
+    local.subnet_newbits,
+    (random_integer.cidr_block.result + count.index) % local.subnet_total,
+  )
+
+  ipv6_cidr_block = cidrsubnet(
+    data.aws_vpc.selected.ipv6_cidr_block,
+    local.subnet_newbits,
+    (random_integer.cidr_block.result + count.index) % local.subnet_total, // TODO: fix this hack
+  )
+
+  assign_ipv6_address_on_creation = true
+
+  tags = tomap({
+    "Name"                   = "${var.cluster_name}-${data.aws_availability_zones.available.names[count.index]}",
+    "Cluster"                = var.cluster_name,
+    (local.kube_cluster_tag) = "shared",
+  })
+}
+
+################################### FIREWALL ###################################
+
+resource "aws_security_group" "common" {
+  name        = "${var.cluster_name}-common"
+  description = "cluster common rules"
+  vpc_id      = data.aws_vpc.selected.id
+
+  tags = tomap({
+    "Cluster"                = var.cluster_name,
+    (local.kube_cluster_tag) = "shared",
+  })
+}
+
+resource "aws_security_group_rule" "ingress_self_allow_all" {
+  type              = "ingress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all incoming traffic from members of this group"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  self        = true
+}
+
+resource "aws_security_group_rule" "egress_allow_all" {
+  type              = "egress"
+  security_group_id = aws_security_group.common.id
+
+  description = "allow all outgoing traffic"
+  from_port   = 0
+  to_port     = 0
+  protocol    = "-1"
+  cidr_blocks = ["0.0.0.0/0"]
```

Comment:
Don't you need some IPv6 firewall rules to allow egress + ingress (nodeports) IPv6 traffic?

Reply:
Hmm yes I think. Lemme check.